code/modules/mob/living/simple_animal/hostile/oldzombie.dm

```diff
@@ -53,20 +54,29 @@
 			else
 				var/infectchance = rand(1,3)
 				if(infectchance == 2)
-					if(H.startinfected == 1)
-						src.health = src.health + 50
-						src << "You just infected <b>[L]</b> for the first time! You have restored 50 HP! And gained <b>3</b> additional infection points!"
-						src.numinfected = src.numinfected + 3
-					src << "You infect <b>[L]!</b> restoring 20 HP!"
-					visible_message("<span class='danger'>[src] bites [L]!</span>")
-					playsound(src.loc, 'sound/weapons/bite.ogg', 50, 1)
-					H.infected = 1
-					H << "<span class='danger'>That bite felt sore as hell! It's getting worse....</span>"
-					H.oldInfect(H)
-					src.numinfected = src.numinfected + 1
-					src << "You gain <b>1</b> infection point!"
-					src << "You now have <b>[src.numinfected]</b> infection points!"
-					src << "<span class='userdanger'>They'll be getting up on their own, just give them a minute!</span>"
+					. = 1 // Default to returning true.
+					if(H.dna && PIERCEIMMUNE in H.dna.species.specflags)
```

Comment:
I would suggest moving the H.dna.species.specflags into another if statement within this one. Even if you're checking H.dna first it could still fail because you're trying to reference H.dna.species...